Source/WebCore/dom/TreeScope.cpp

```diff
@@ -456,8 +457,8 @@ Element* TreeScope::findAnchor(StringView name)
 {
     if (name.isEmpty())
         return nullptr;
-    if (Element* element = getElementById(name))
-        return element;
+    if (RefPtr element = getElementById(name))
+        return element.get();
```

Comment:
I would have used `auto` here, not `RefPtr`.

But also, this doesn’t look safe. Why can we just return a raw pointer and let the `RefPtr` fall out of scope? That seems intrinsically dangerous. Seems like `findAnchor` needs to return a `RefPtr<Element>`.

Reply:
For things that return a RefPtr / Ref, I much prefer using `RefPtr` / `Ref` type for the local variable than `auto`. It makes it much clearer that the new code is safe and it makes the effort to adopt smart pointers easier. Prior to C++17, where we had to specify `RefPtr<TYPE>` then it would have been annoying. However, now that we can just use `RefPtr / Ref` without specifying the underlying type, I don't see any big drawback to using them explicitly.

In terms of safety, I believe the code is safe. getElementById() is not creating an element, it is returning an existing one. It happens to return a RefPtr<> because it was holding a RefPtr internally anyway so there was no cost to returning a RefPtr. This also reduces ref counting churn as call sites will often store the result in a RefPtr anyway.

We may want findAnchor() to return a RefPtr still. I will look into it. As I remember it, one of the return sites didn't already have a RefPtr to return so it would have created new ref counting churn to change the return type. My policy has been to only update the return type to be a RefPtr if there is no extra cost to do so (i.e. all return sites already had a RefPtr). Otherwise, our smart pointer adoption policy doesn't require changing any return types or argument types AFAIK.